### PR TITLE
SNOW-693548 fixing redos attack

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-693548

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This fixes the CVE https://nvd.nist.gov/vuln/detail/CVE-2022-42965
   The issue comes from the greediness of the original regex. With this reworked function we have roughly linear performance. This is achieved by starting from the beginning to lazily strip away comments one-by-one instead of matching the whole SQL command with one regex. The new SQL is also better by matching more than multi-line comments.

### CODE REVIEW

**Comment by sfc-gh-mkeller** [comment]
So as it turns out the new Regex is now able to find more comments and this has broken [this old test](https://github.com/snowflakedb/snowflake-connector-python/blob/aa6c88f5a0f72feb3a2fa0329b6b253bb52e2394/test/integ/test_put_get_medium.py#L761).
Which makes this definitely a behavior change 😭 I've modified the old test in [2faa971](https://github.com/snowflakedb/snowflake-connector-python/pull/1327/commits/2faa9711272c4ba2a2834663405202e93ff907d3).

File: src/snowflake/connector/_sql_util.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import re
+
+from .constants import FileTransferType
+
+COMMENT_START_SQL_RE = re.compile(
+    r"""
+                                  ^\s*(?:
+                                      /\*[\w\W]*?\*/
+                                  )""",
```
Comment (sfc-gh-sfan) [inline src/snowflake/connector/_sql_util.py:15]: Why do we need these new lines and spaces in this regex?

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/_sql_util.py:15]: It's easier to read. Here's the doc on [re.VERBOSE](https://docs.python.org/3/library/re.html#re.VERBOSE)

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-sfan — APPROVED** [review]

**Review by sfc-gh-ssarangpur — COMMENTED** [review]
LGTM

**Comment by sfc-gh-hpathak** [comment]
@sfc-gh-mkeller  Looks good overall for the CVE. However, I wonder if there is a potential to exploit this again via some other means. We may want to think about making this robust at a design level.

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**